cpp/oneapi/dal/detail/memory_impl_host.cpp

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/detail/memory_impl_host.hpp"
+
+#include <daal/include/services/daal_memory.h>
+
+namespace oneapi::dal::detail::v1 {
+
+template <typename AllocOp>
+void* alloc_impl(AllocOp&& op, std::size_t size, std::size_t alignment) {
+    auto ptr = op(size, alignment);
+    if (ptr == nullptr) {
+        throw dal::host_bad_alloc();
+    }
+    return ptr;
+}
+
+void* malloc(const default_host_policy&, std::size_t size) {
```

Comment:
I think, we'll need to deprecate `default_host_policy` and always use normal `host_policy`. This can be done in other PR.